### PR TITLE
Button, Icon: fix iconSize prop doesn't work with some icons

### LINKED ISSUE #43745: Button: `iconSize` prop doesn't work with some icons
### Description

I have found that the size is not applied to some of the `@wordpress/icons` when I specify the `iconSize` attribute on a button component with SVG icons.
These icons will be displayed at the expected size when the size is specified in the `Icons` component, not the `Button` component.

I checked the icon library and found that icons with `width` / `height` attributes of the svg element match those with no size applied. I don't know why these icons don't reproduce when used in the Icons component, but I think SVG icons should be rendered at the size specified in the `Icons` component and the `width` / `height` attributes are unnecessary on the svg element.

The following are those icons:

- [atSymbol](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/at-symbol.js)
- [commentEditLink](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/comment-edit-link.js)
- [commentReplyLink](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/comment-reply-link.js)
- [dragHandle](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/drag-handle.js)
- [helpFilled](https://github.com/WordPress/gutenberg/blob/trunk/packages/icons/src/library/help-filled.js)

### Step-by-step reproduction instructions

I have rewritten `edit.js` in the code block to include code that designates all icons in `@wordpress/icons` as buttons:

```javascript
import { useBlockProps } from '@wordpress/block-editor';
import * as Icons from '@wordpress/icons';
import { Button, Flex } from '@wordpress/components';

const { Icon, ...availableIcons } = Icons;

export default function CodeEdit() {
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<Flex
				wrap="wrap"
				justify="flex-start"
				style={ { textAlign: 'center', fontSize: '8px' } }
			>
				{ Object.entries( availableIcons ).map( ( [ name, icon ] ) => {
					return (
						<div key={ name } style={ { width: '5%' } }>
							<Button icon={ icon } iconSize={ 40 } />
						</div>
					);
				} ) }
			</Flex>
		</div>
	);
}
```

### Screenshots, screen recording, code snippet

The icon circled in red has `width` and `height` specified for the svg element and the `iconSize` property is not applied.

icons

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fix #43745

## What?
~~This PR fixes an issue where the `iconSize` prop of the `Button` component was not reflected in some icons.~~
This PR fixes a problem in the `@wordpress/components` package where the `iconSize` prop of the `Button` and `Icon` component was not reflected in some icons.

## Why?
This is because the `iconSize` prop is overridden if the icon's svg element has a `width` / `height` attribute, as noted in [this comment](https://github.com/WordPress/gutenberg/pull/43747#pullrequestreview-1093433535).

The following are those icons:

- [atSymbol](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/at-symbol.js)
- [commentEditLink](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/comment-edit-link.js)
- [commentReplyLink](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/comment-reply-link.js)
- [dragHandle](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/drag-handle.js)
- [helpFilled](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/help-filled.js)

## How?

In the internal Icon component, I have made the passed props (such as iconSize) take precedence over the attributes of the SVG element.

## Testing Instructions
Rewrite edit.js for any of the core blocks as follows and confirm that they all render at the expected size:

_@add:_ This example also uses `@wordpress/icons` Icon, which is incorrect. It's not a problem with the `@wordpress/icons` package, as [this comment states](https://github.com/WordPress/gutenberg/pull/43821#pullrequestreview-1096509151).

```javascript
/**
 * WordPress dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';
import { Button } from '@wordpress/components';
import { Icon, help, helpFilled } from '@wordpress/icons';

export default function CodeEdit() {
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<p>Icon without `size` prop (24px expected)</p>
			<Icon icon={ help } />
			<Icon icon={ helpFilled } />

			<p>Icon with `size` prop (60px expected)</p>
			<Icon icon={ help } size="60" />
			<Icon icon={ helpFilled } size="60" />

			<p>Button without `iconSize` prop (24px expected)</p>
			<Button icon={ help } />
			<Button icon={ helpFilled } />

			<p>Button with `iconSize` prop (60px expected)</p>
			<Button icon={ help } iconSize="60" />
			<Button icon={ helpFilled } iconSize="60" />
		</div>
	);
}
```

- The `help` icon on the left has no width / height attributes on the svg element.
- The `helpFilled` icon on the right has width / height attributes on the svg element.

### On trunk

on-trunk

### ON this branch

on-branch

### CODE REVIEW

**Comment by t-hamano** [comment]
This PR has changed the order of the attributes of the `svg` element in some tests, but I believe there is no problem at all.

File: packages/components/src/icon/test/index.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ describe( 'Icon', () => {
 		expect( wrapper.prop( 'height' ) ).toBe( 24 );
 	} );
 
-	it( 'renders an svg element and passes the size as its width and height', () => {
+	it( 'renders an svg element and override its width and height', () => {
```
Comment (t-hamano) [inline packages/components/src/icon/test/index.js:66]: This test is originally intended to test that the width/height of the svg element is respected, but should be overridden by the `size` prop of the `Icon` component.

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -4,12 +4,13 @@
 
 ### Bug Fix
 
+-   `Button`: Fix `iconSize` prop doesn't work with some icons ([#43821](https://github.com/WordPress/gutenberg/pull/43821)).
```
Comment (mirka) [inline packages/components/CHANGELOG.md]: We probably also want to mention that it was fixed in the `Icon` component.

**Review by mirka — APPROVED** [review]
Works great, thanks 👍 

Just to clarify so others don't get confused by the PR description: This fix is in the `@wordpress/components` Icon component, **not** the `@wordpress/icons` Icon. The wp-icons Icon component already works as expected on trunk.

**Comment by t-hamano** [comment]
> Just to clarify so others don't get confused by the PR description: This fix is in the @wordpress/components Icon component, not the @wordpress/icons Icon. The wp-icons Icon component already works as expected on trunk.

Thanks for pointing that out! The sample code I presented contains the wrong test 😅
I have updated the title and description so that people who see this PR will not be misleading.

Is this title appropriate as a commit message at merge time?

File: packages/components/CHANGELOG.md (hunk 2 above)
Comment (t-hamano) [reply packages/components/CHANGELOG.md]: Fixed by 6c4d6a3

**Comment by mirka** [comment]
> Is this title appropriate as a commit message at merge time?

Works for me 👍

**Pull request merged**